src/EventStore.Core/ProcessTracker.cs

```diff
@@ -0,0 +1,190 @@
+		_eventCountersHelper = new EventCountersHelper(2_500);
+	}
+
+	private T Bookkeeping<T>(Func<T> func) {
+		var now = _clock.Now;
+		if (_lastTime != Instant.FromSeconds(0) && now.ElapsedSecondsSince(_lastTime) < 5)
+			return func();
+
+		_lastTime = now;
+
+		if (OS.IsUnix) {
+			_loadAverages = _stats.GetLoadAverages();
+		}
+
+		_esDriveInfo = EsDriveInfo.FromDirectory(_dbPath, _log);
+		_diskIo = DiskIo.GetDiskIo(Environment.ProcessId, _log);
+		_process = Process.GetCurrentProcess();
+
+		return func();
+	}
+
+	public double CollectCpuAvg1M() {
+		return Bookkeeping(() => _loadAverages.Average1m);
+	}
+
+	public double CollectCpuAvg5M() {
+		return Bookkeeping(() => _loadAverages.Average5m);
+	}
+
+	public double CollectCpuAvg15M() {
+		return Bookkeeping(() => _loadAverages.Average15m);
+	}
+
+	public float CollectCpuTotalUsage() {
+		return Bookkeeping(() => _perfCounter.GetTotalCpuUsage());
+	}
+
+	public long CollectFreeMemoryUnix() {
+		return Bookkeeping(() => (long)_stats.GetFreeMemory());
+	}
+
+	public long CollectFreeMemoryWindows() {
+		return Bookkeeping(() => _perfCounter.GetFreeMemory());
+	}
+
+	public long CollectTotalMemoryUnix() {
+		return Bookkeeping(() => (long)_stats.GetTotalMemory());
+	}
+
+	public long CollectTotalMemoryWindows() {
+		return Bookkeeping(() => (long)WinNativeMemoryStatus.GetTotalMemory());
+	}
+
+	public Measurement<double> CollectProcCPU() {
+		return Bookkeeping(() => {
+			var dim = new KeyValuePair<string, object?>[] {
+				new("id", _process!.Id),
+				new("mem", _process!.WorkingSet64),
+				new("startTime", _process!.StartTime.ToUniversalTime().ToString("O")),
+				new("threadsCount", ThreadPool.ThreadCount),
+				new("contentionRate", Monitor.LockContentionCount),
+				new("thrownExceptionsRate", _eventCountersHelper.GetThrownExceptionsRate()),
+			};
+
+			return new Measurement<double>(ProcCpuUsage(), dim);
+		});
+	}
+
+	public long CollectDriveIOReadBytes() {
+		return (long)Bookkeeping(() => _diskIo?.ReadBytes ?? 0);
+	}
+
+	public long CollectDriveIOReadOps() {
+		return (long)Bookkeeping(() => _diskIo?.ReadOps ?? 0);
+	}
+
+	public long CollectDriveIOWriteBytes() {
+		return (long)Bookkeeping(() => _diskIo?.WrittenBytes ?? 0);
+	}
+
+	public long CollectDriveIOWriteOps() {
+		return (long)Bookkeeping(() => _diskIo?.WriteOps ?? 0);
+	}
+
+	public long CollectDriveTotalBytes() {
+		return Bookkeeping(() => _esDriveInfo?.TotalBytes ?? 0);
+	}
+
+	public long CollectDriveUsedBytes() {
+		return Bookkeeping(() => _esDriveInfo?.UsedBytes ?? 0);
+	}
+
+	public long CollectGcAllocationSpeed() {
+		return Bookkeeping(() => GC.GetTotalAllocatedBytes());
+	}
+
+	public double CollectGcFragmentation() {
+		return Bookkeeping(() => {
+			var info = GC.GetGCMemoryInfo();
+			return info.HeapSizeBytes != 0 ? info.FragmentedBytes * 100d / info.HeapSizeBytes : 0;
+		});
+	}
+
+	public long CollectGc0Size() {
+		return Bookkeeping(() => _eventCountersHelper.GetGcStats().Gen0Size);
```

Comment:
would prefer to have a specific method on EventCountersHelper for each of these (like GetThrownExceptionsRate already has). As it is, GetGcStats() will get 11 counter values as well as allocating the GcStats object, and then we'll be doing this for each of CollectGc0Size/CollectGc1Size/etcetc

Reply:
(not necessarily now that we are not using eventcountershelper at all)